### PR TITLE
add ability to set model to playermodel

### PR DESCRIPTION
**Pull request by 2dve**

### CODE REVIEW

**Comment by CapsAdmin** [comment]
Can you explain the need for this? I think I would do it some other way if I understand the need correctly.

You either want the model to reflect the local player's model, meaning if I'm kleiner the model will be kleiner, but if someone is alyx the model will be alyx for them, but kleiner for me.

Or you want the model to reflect the outfit's player model.

**Comment by 2dve** [comment]
Thank you, this has been fixed. The feature is to set the model to the outfit wearer's playermodel. I used this to achieve the effect of replacing the right arm, which would have been impossible otherwise due to the fact that weapons scale with the right hand bone.

**Comment by 2dve** [comment]
basically instead of having to manually use events and custom weapon models to make the weapons appear, i just have it set up as pictured with the playermodel being invisible and the weapon still showing
image
image

**Comment by 2dve** [comment]
This feature can also make it easier for server developers to add items that require the player model to be used, as there is no way currently for PAC to automatically change a model to the current player model.

**Comment by CapsAdmin** [comment]
I think instead of doing this, I would make it so you can select "player_model" as a model in the model field.

File: lua/pac3/core/client/parts/model.lua [hunk 1]
```diff
@@ -405,6 +406,16 @@ function PART:DrawModel(ent, pos, ang)
 	if self.loading and not self.obj_mesh then return end
 
 
+	if self.SetModelToPlayer then
```
Comment (thegrb93) [inline lua/pac3/core/client/parts/model.lua:409]: Doing this every frame is not good for performance

**Pull request closed without merging**